python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -185,13 +167,21 @@ def evaluate(
         df_grouped_stats = pd.DataFrame()
         # defining variables to compute running mean and variance for t-test computations
         agg_count, agg_mean, agg_M2 = 0, 0, 0
+        group_metric_running_variance_params = {}
+        group_key = list(set(self.feature_config.get_group_metrics_keys("node_name")))
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
-            # Accumulating statistics for t-test calculation using 1/rank
+            # Add necessary columns for power analysis
             clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
-            diff = (1/clicked_records[RankingConstants.NEW_RANK] - 1/clicked_records[
-                self.feature_config.get_rank("node_name")]).to_list()
+            clicked_records[RankingConstants.NEW_MRR] = 1.0 / clicked_records["new_rank"]
+            clicked_records[RankingConstants.OLD_MRR] = 1.0 / clicked_records[self.feature_config.get_rank("node_name")]
+            # Statistical analysis pre-processing
+            if len(group_key) > 0:
+                statistical_analysis_preprocessing(clicked_records, group_metric_running_variance_params, group_key,
+                                                   RankingConstants.VARIANCE_METRIC_LIST)
+
+            diff = (clicked_records[RankingConstants.NEW_MRR] - clicked_records[RankingConstants.OLD_MRR]).to_list()
```

Comment:
@mohazahran Isn't MRR and ACR already being computed in the `metrics_helper`? Why do we need to do it here again?